# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBCCodegen-inl.h
# Commit: 3ecda60613c5
# Full Hash: 3ecda60613c5cb7a57cc7004919c7d12c3a29268
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-22 03:08:27
# Description:
#   Bug 1736531 - Fix. r=jseward
#   
#   See bug for analysis.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D129010
# ==============================================================================

diff -r 1044b04cf89b -r 3ecda60613c5 js/src/wasm/WasmBCCodegen-inl.h
--- a/js/src/wasm/WasmBCCodegen-inl.h	Thu Oct 21 15:44:59 2021 +0000
+++ b/js/src/wasm/WasmBCCodegen-inl.h	Thu Oct 21 15:46:28 2021 +0000
@@ -138,10 +138,8 @@
   RegI32 r = RegI32(ReturnReg);
   MOZ_ASSERT(isAvailableI32(r));
   needI32(r);
-#if defined(JS_CODEGEN_X64)
-  if (JitOptions.spectreIndexMasking) {
-    masm.movl(r, r);
-  }
+#if defined(JS_64BIT)
+  masm.widenInt32(r);
 #endif
   return r;
 }